<a href="https://colab.research.google.com/github/sierraolmoss-ui/prog_civil_-2026_1/blob/main/Copia_de_Copia_de_Copia_de_2_3_4_Actividad_Pluma_de_Agua_Algoritmo_GitHub.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# NOMBRES COMPLETOS: Saed Vanessa Sierra Olmos
# Código: 1103500979

# Actividad de Programación con Python  
## **Historia, ciencia y programación: la “pluma de agua”**

> **Relato de partida**  
> **¿Por qué los costeños le decimos “pluma” a la llave del agua?**  
>  
> En el Caribe colombiano, abrir la “pluma” no es solo una acción cotidiana: es una expresión cargada de historia, técnica y cultura. Aunque algunos se sorprendan o incluso se burlen del término, lo cierto es que decir “pluma” en vez de “llave” no es un error, sino una herencia viva que conecta nuestro lenguaje con siglos de evolución hidráulica.  
>  
> Mucho antes de que existieran los grifos modernos, el agua se distribuía mediante sistemas regulados con precisión. En España, por ejemplo, se usaba la **“pluma de agua”** como unidad de medida para calcular el caudal que se entregaba a cada hogar. Aproximadamente, una pluma equivalía a **0.025 litros por segundo**. En algunos lugares, ese flujo se regulaba mediante orificios calibrados y dispositivos llamados **plumeros**, que permitían controlar la salida del agua.  
>  
> Durante la época colonial, muchas de estas prácticas técnicas cruzaron el Atlántico y llegaron a ciudades como Cartagena, Barranquilla y Santa Marta. Con el tiempo, el término “pluma” dejó de ser solamente técnico y pasó al habla cotidiana como sinónimo de la llave del agua.  
>  
> Así, una palabra del uso diario conserva una memoria histórica, tecnológica y cultural.  
>  
> ---
>  
> **Idea central para esta actividad:**  
> A partir de este relato, construiremos un problema de programación donde una magnitud histórica y física, el **caudal de una pluma de agua**, se represente mediante variables, operaciones y estructuras básicas de Python.

## 1. Propósito de la actividad

Al finalizar esta actividad, el estudiante debe ser capaz de:

- interpretar un contexto histórico y científico como un problema computacional;
- identificar entradas, procesos y salidas;
- diseñar un algoritmo antes de programar;
- implementar el programa en Python;
- publicar el trabajo en GitHub;
- ampliar la documentación del código;
- clonar el repositorio al disco local usando Git.

> **Importante:**  
> En este notebook **no se entrega la solución completa**.  
> Aquí se presenta el **planteamiento**, la **estructura del problema** y el **algoritmo guía** para que cada estudiante construya el código.

## 2. Situación problema

Un hogar recibe una cierta cantidad de **plumas de agua**.  
Se sabe que:

- **1 pluma = 0.025 litros por segundo**

Con esta información, el estudiante debe construir un programa que permita calcular:

1. el caudal total en litros por segundo;
2. el consumo diario en litros;
3. el consumo mensual en litros (suponiendo 30 días);
4. una clasificación del consumo:
   - **Bajo**: menos de 1000 litros por día
   - **Medio**: entre 1000 y 3000 litros por día
   - **Alto**: más de 3000 litros por día

---

## 3. Análisis del problema

### 3.1 Entrada del sistema
El programa debe recibir:

- número de plumas asignadas al hogar.

### 3.2 Proceso
El programa debe:

- convertir el número de plumas en caudal total;
- transformar el caudal a consumo diario;
- transformar el consumo diario a consumo mensual;
- clasificar el nivel de consumo.

### 3.3 Salida
El programa debe mostrar:

- caudal total;
- consumo diario;
- consumo mensual;
- clasificación del consumo.

## 4. Variables y constantes sugeridas

El estudiante puede trabajar con variables como las siguientes:

- `plumas`
- `caudal_pluma`
- `caudal_total`
- `segundos_dia`
- `consumo_diario`
- `consumo_mensual`
- `nivel_consumo`

> Puede cambiar los nombres, pero deben ser claros y significativos.

## 5. Algoritmo propuesto

Antes de programar, escriba el algoritmo en lenguaje natural o seudocódigo.

### Algoritmo en lenguaje natural

1. Solicitar al usuario el número de plumas de agua.
2. Definir el valor de una pluma en litros por segundo.
3. Definir la cantidad de segundos que tiene un día.
4. Calcular el caudal total.
5. Calcular el consumo diario.
6. Calcular el consumo mensual.
7. Determinar si el consumo es bajo, medio o alto.
8. Mostrar los resultados de manera clara.

### Seudocódigo guía

```text
Inicio
    Leer número de plumas
    Definir caudal de una pluma
    Definir segundos por día

    Calcular caudal total
    Calcular consumo diario
    Calcular consumo mensual

    Si consumo diario es menor que 1000
        nivel = "Bajo"
    Si no, si consumo diario está entre 1000 y 3000
        nivel = "Medio"
    En otro caso
        nivel = "Alto"

    Mostrar caudal total
    Mostrar consumo diario
    Mostrar consumo mensual
    Mostrar nivel
Fin
```

## 6. Espacio para que el estudiante construya la solución

### Instrucciones
Complete el siguiente bloque de código a partir del algoritmo anterior.

- No copie una solución terminada sin comprenderla.
- Escriba comentarios propios.
- Use nombres de variables comprensibles.
- Verifique que los resultados tengan sentido físico.

In [ ]:
# ==========================================================
# ACTIVIDAD: MODELAR LA "PLUMA DE AGUA" CON PYTHON
# ==========================================================

# ENTRADA :Solicitar al usuario el número de plumas
plumas = float(input("Ingrese el número de plumas: "))

# PROCESO : Calcular el caudal total , consumo diario y consumo mensual

##Definir constantes del problema

Caudal_pluma= 0.025 # litros por segundo

segundos_dia= 24 * 60 * 60 # segundos

dias_mes= 30 # dias

##Calcular el caudal total de agua que fluye a través de todas las plumas asignadas.

caudal_total= plumas * Caudal_pluma # litros/segundo

##Calcular el consumo diario

consumo_diario= caudal_total * segundos_dia # litros/dia

##Calcular el consumo mensual

consumo_mensual= consumo_diario * dias_mes # litros/mes

## Clasificar el consumo
nivel_consumo=""

if consumo_diario < 1000:
  nivel_consumo = "Bajo"
elif 1000 <= consumo_diario <= 3000:
  nivel_consumo = "Medio"
else:
  nivel_consumo = "Alto"

# SALIDA: imprimir el resultado

## Mostrar resultados
print(f" Resultados del Consumo de Agua ")
print(f"Número de plumas ingresadas: {plumas}")
print(f"Caudal total: {caudal_total:.3f} litros por segundo")
print(f"Consumo diario: {consumo_diario:.2f} litros")
print(f"Consumo mensual: {consumo_mensual:.2f} litros")
print(f"Clasificación del consumo: {nivel_consumo}")


Ingrese el número de plumas: 8
 Resultados del Consumo de Agua 
Número de plumas ingresadas: 8.0
Caudal total: 0.200 litros por segundo
Consumo diario: 17280.00 litros
Consumo mensual: 518400.00 litros
Clasificación del consumo: Alto


## 7. Verificación conceptual

Antes de dar por terminado el programa, responda en comentarios o en una celda Markdown:

1. ¿Qué representa físicamente la variable principal del problema?
2. ¿Por qué este ejercicio conecta historia, cultura e ingeniería?
3. ¿Qué operaciones aritméticas fueron necesarias?
4. ¿Qué estructura condicional utilizó y por qué?
5. ¿Cómo interpretaría el resultado en un contexto real?

## Respuestas a la Verificación Conceptual

1.  **¿Qué representa físicamente la variable principal del problema?**
    La variable principal `plumas` representa físicamente el número de unidades de 'plumas de agua' asignadas a un hogar, que es una medida histórica de caudal. Cada pluma equivale a un caudal específico en litros por segundo.

2.  **¿Por qué este ejercicio conecta historia, cultura e ingeniería?**
    Este ejercicio conecta historia al usar el término 'pluma de agua' que proviene de prácticas coloniales. Conecta cultura al reconocer que en el Caribe colombiano se usa 'pluma' para referirse a la llave del agua, una herencia de esa historia. Y conecta ingeniería al modelar un concepto físico (caudal) y sus transformaciones (consumo diario/mensual) usando cálculos y lógica programática, que son principios de ingeniería.

3.  **¿Qué operaciones aritméticas fueron necesarias?**
    Se utilizaron las siguientes operaciones aritméticas:
    *   **Multiplicación (`*`)**: Para calcular el caudal total (plumas * caudal_pluma), el consumo diario (caudal_total * segundos_dia) y el consumo mensual (consumo_diario * dias_mes).
    *   **División (`/`)**: Implícitamente en la definición de `Caudal_pluma` (litros por segundo).
    *   **Suma (`+`)**: No se usó directamente para los cálculos principales, pero podría ser parte de operaciones más complejas.

4.  **¿Qué estructura condicional utilizó y por qué?**
    Se utilizó una estructura condicional `if-elif-else` para clasificar el nivel de consumo (`nivel_consumo`). Esta estructura es ideal cuando se necesita evaluar múltiples condiciones mutuamente excluyentes (bajo, medio, alto) y ejecutar un bloque de código específico para la primera condición que se cumpla. Permite una lógica clara y organizada para la toma de decisiones basada en el valor de `consumo_diario`.

5.  **¿Cómo interpretaría el resultado en un contexto real?**
    En un contexto real, los resultados nos dan una idea cuantitativa del uso del agua. Si el `consumo_diario` es alto (como 17280 litros para 8 plumas), esto podría indicar un uso excesivo de agua, una fuga, o que la asignación de plumas es demasiado alta para un hogar promedio. Esta información sería crucial para:
    *   **Gestión de recursos:** Las empresas de servicios públicos podrían usar esto para monitorear el consumo.
    *   **Conciencia del usuario:** Los hogares podrían ajustar sus hábitos si ven que su consumo es 'Alto'.
    *   **Políticas de ahorro:** Podría informar decisiones sobre tarifas o programas de conservación.

    El valor 'Alto' para un consumo diario de 17,280 litros sugiere un consumo muy elevado que, para un hogar, sería insostenible o indicaría una anomalía.

## 8. Extensión opcional

Si desea profundizar, puede ampliar el programa para que también permita:

- calcular el consumo semanal;
- comparar el consumo de varios hogares;
- estimar el costo del agua si se conoce una tarifa por litro o por metro cúbico;
- convertir litros a metros cúbicos;
- presentar los resultados en forma tabular.

> Estas extensiones no reemplazan la solución base; son una ampliación.

## 9. Documentación del código con apoyo de IA

Después de construir y probar su programa:

1. Copie su código en ChatGPT.
2. Solicite una **explicación técnica en Markdown** del código.
3. Pida además:
   - descripción de la función del programa;
   - explicación de cada sección;
   - explicación de variables, entradas, procesos y salidas;
   - observaciones sobre claridad y buenas prácticas.
4. Mejore su código con base en esa retroalimentación.
5. Actualice su repositorio en GitHub con la versión mejorada y documentada.

### Prompt sugerido para documentar el código

```text
Actúa como docente universitario experto en programación con Python.

Te voy a compartir un código en Python.
Necesito que expliques el código en formato Markdown de manera clara, técnica y ordenada.

Incluye:
1. propósito general del programa;
2. explicación de entradas, procesos y salidas;
3. descripción de cada variable importante;
4. explicación de las estructuras usadas;
5. observaciones sobre claridad, estilo y posibles mejoras;
6. una versión comentada del código, sin cambiar su lógica.

Aquí está mi código:
[PEGAR AQUÍ EL CÓDIGO]
```

## Documentación y Análisis del Código: Modelo de la "Pluma de Agua"

---

### 1. Propósito General del Programa

Este programa tiene como objetivo principal simular y calcular el consumo de agua basado en el concepto histórico de la "pluma de agua", una antigua unidad de medida de caudal. A partir del número de plumas asignadas a un hogar, el programa calcula el caudal total en litros por segundo, el consumo diario y mensual en litros, y clasifica el nivel de consumo (Bajo, Medio, Alto) basándose en umbrales predefinidos. Este ejercicio conecta conceptos históricos con su representación computacional, demostrando cómo variables y operaciones básicas de Python pueden modelar fenómenos del mundo real.

### 2. Explicación de Entradas, Procesos y Salidas

*   **Entrada:**
    *   El programa solicita al usuario un único dato de entrada: el `número de plumas` asignadas, que representa la capacidad de flujo de agua del hogar. Se espera un valor numérico, que puede ser decimal.

*   **Proceso:**
    1.  **Definición de Constantes:** Se establecen valores fijos para el caudal de una pluma (`Caudal_pluma`), la cantidad de segundos en un día (`segundos_dia`) y los días en un mes (`dias_mes`).
    2.  **Cálculo del Caudal Total:** Multiplica el número de plumas ingresado por el caudal de una pluma para obtener el caudal total en litros por segundo.
    3.  **Cálculo del Consumo Diario:** Transforma el caudal total (litros/segundo) a consumo en litros por día, multiplicando por la cantidad de segundos en un día.
    4.  **Cálculo del Consumo Mensual:** Calcula el consumo total en litros para un mes, multiplicando el consumo diario por la cantidad de días en un mes.
    5.  **Clasificación del Consumo:** Evalúa el `consumo_diario` mediante una estructura condicional (`if-elif-else`) para determinar si el consumo se clasifica como "Bajo", "Medio" o "Alto".

*   **Salidas:**
    *   El programa imprime de manera clara y formatada los siguientes resultados:
        *   El número de plumas ingresadas por el usuario.
        *   El `caudal_total` en litros por segundo, con tres decimales.
        *   El `consumo_diario` en litros, con dos decimales.
        *   El `consumo_mensual` en litros, con dos decimales.
        *   La `Clasificación del consumo` (Bajo, Medio, Alto).

### 3. Descripción de Cada Variable Importante

*   `plumas` (float): Almacena el número de plumas de agua que el usuario ingresa. Es la variable principal de entrada.
*   `Caudal_pluma` (float): Constante que define el caudal de una sola pluma en litros por segundo (0.025 L/s).
*   `segundos_dia` (int): Constante que representa la cantidad total de segundos en un día (24 * 60 * 60 = 86400 segundos).
*   `dias_mes` (int): Constante que define la cantidad de días considerados en un mes para el cálculo mensual (30 días).
*   `caudal_total` (float): Resultado del cálculo del caudal combinado de todas las plumas en litros por segundo.
*   `consumo_diario` (float): Representa el total de litros consumidos en un día, derivado del `caudal_total`.
*   `consumo_mensual` (float): Indica el total de litros consumidos en un mes, basado en el `consumo_diario`.
*   `nivel_consumo` (str): Almacena la clasificación del consumo de agua ("Bajo", "Medio", "Alto") según el `consumo_diario`.

### 4. Explicación de las Estructuras Usadas

*   **Función `input()`:** Utilizada para obtener la entrada del usuario (`plumas`).
*   **Función `float()`:** Convierte la entrada de texto del usuario a un número de punto flotante, permitiendo el uso de valores decimales para las plumas.
*   **Operadores Aritméticos:** Multiplicación (`*`) para realizar todos los cálculos de caudal y consumo.
*   **Estructura Condicional `if-elif-else`:** Empleada para la lógica de clasificación del consumo. Permite evaluar diferentes rangos para `consumo_diario` y asignar el `nivel_consumo` correspondiente. Es adecuada para situaciones donde hay múltiples condiciones mutuamente excluyentes.
*   **Función `print()` con f-strings:** Utilizada para mostrar los resultados de forma clara y legible. Los f-strings (`f"..."`) facilitan la inserción de variables y el formato de los números flotantes con un número específico de decimales (ej. `:.2f`).

### 5. Observaciones sobre Claridad, Estilo y Posibles Mejoras

**Claridad y Estilo:**
*   **Nombres de Variables:** Los nombres de variables como `Caudal_pluma`, `segundos_dia`, `consumo_diario` son claros y descriptivos, lo cual mejora la legibilidad del código.
*   **Comentarios:** El código está bien comentado, separando claramente las secciones de ENTRADA, PROCESO y SALIDA, y explicando las constantes y los cálculos. Esto es una muy buena práctica.
*   **Organización:** La estructura del código sigue un flujo lógico de entrada, procesamiento y salida, lo que facilita su comprensión.
*   **Formato de Salida:** El uso de f-strings para formatear la salida es excelente, haciendo que los resultados sean fáciles de leer y entender, especialmente con el redondeo de decimales.

**Posibles Mejoras:**
*   **Encapsulamiento en Funciones:** Para hacer el código más modular y reutilizable, los cálculos y la lógica de clasificación podrían encapsularse en una o varias funciones. Por ejemplo, una función `calcular_consumo(num_plumas)` que devuelva un diccionario con todos los resultados.
*   **Validación de Entrada:** Actualmente, el programa asume que el usuario siempre ingresará un número válido. Se podría añadir un bloque `try-except` para manejar posibles errores si el usuario ingresa texto en lugar de un número, o una validación para asegurar que el número de plumas sea positivo.
*   **Constantes Globales:** Aunque están bien definidas, para programas más grandes, las constantes podrían agruparse en un objeto o un módulo de configuración.
*   **Flexibilidad de Días en el Mes:** El `dias_mes` está fijado en 30. Podría ser una entrada del usuario o derivarse de la librería `datetime` para mayor precisión si se considerara un mes específico.
*   **Interfaz de Usuario:** Para una experiencia más interactiva, se podría considerar una interfaz gráfica sencilla (ej. `tkinter`, `PyQt`) o un sistema de bucle para calcular múltiples escenarios sin reiniciar el programa.
*   **Umbrales de Clasificación:** Los umbrales de 1000 y 3000 litros podrían definirse como constantes al inicio para una fácil modificación.

### 6. Versión Comentada del Código (Sin Cambiar su Lógica)

```python
# ==========================================================
# ACTIVIDAD: MODELAR LA "PLUMA DE AGUA" CON PYTHON
# ==========================================================

# ENTRADA: Solicitar al usuario el número de plumas
# Se pide al usuario que ingrese la cantidad de 'plumas de agua' que tiene un hogar.
# La entrada se convierte inmediatamente a tipo float para permitir valores decimales.
plumas = float(input("Ingrese el número de plumas: "))

# PROCESO: Calcular el caudal total, consumo diario y consumo mensual, y clasificar el consumo.

## Definir constantes del problema
# Caudal de una única pluma en litros por segundo (L/s).
Caudal_pluma = 0.025
# Cantidad total de segundos en un día (24 horas * 60 minutos/hora * 60 segundos/minuto).
segundos_dia = 24 * 60 * 60
# Número de días considerado para el cálculo del consumo mensual.
dias_mes = 30

## Calcular el caudal total de agua que fluye a través de todas las plumas asignadas.
# Multiplica el número de plumas ingresado por el caudal de una pluma individual.
caudal_total = plumas * Caudal_pluma # litros/segundo

## Calcular el consumo diario
# Convierte el caudal total (L/s) a consumo diario (L/día) multiplicando por los segundos en un día.
consumo_diario = caudal_total * segundos_dia # litros/dia

## Calcular el consumo mensual
# Calcula el consumo total en litros para un mes, basado en el consumo diario y los días del mes.
consumo_mensual = consumo_diario * dias_mes # litros/mes

## Clasificar el consumo
# Inicializa la variable que contendrá la clasificación del consumo.
nivel_consumo = ""

# Utiliza una estructura condicional if-elif-else para clasificar el consumo diario.
# Si el consumo diario es menor de 1000 litros.
if consumo_diario < 1000:
  nivel_consumo = "Bajo"
# Si el consumo diario está entre 1000 y 3000 litros (inclusive).
elif 1000 <= consumo_diario <= 3000:
  nivel_consumo = "Medio"
# Si el consumo diario es mayor de 3000 litros.
else:
  nivel_consumo = "Alto"

# SALIDA: Imprimir los resultados calculados al usuario.

## Mostrar resultados
# Imprime un encabezado para los resultados.
print(f" Resultados del Consumo de Agua ")
# Muestra el número de plumas ingresado por el usuario.
print(f"Número de plumas ingresadas: {plumas}")
# Muestra el caudal total, formateado a tres decimales.
print(f"Caudal total: {caudal_total:.3f} litros por segundo")
# Muestra el consumo diario, formateado a dos decimales.
print(f"Consumo diario: {consumo_diario:.2f} litros")
# Muestra el consumo mensual, formateado a dos decimales.
print(f"Consumo mensual: {consumo_mensual:.2f} litros")
# Muestra la clasificación del consumo determinada.
print(f"Clasificación del consumo: {nivel_consumo}")
```

## 10. Trabajo con GitHub

Una vez terminado el programa:

1. Cree un repositorio en GitHub.
2. Suba el notebook o el archivo `.py`.
3. Haga un primer commit.
4. Después de ampliar la documentación del código, haga un segundo commit.
5. Copie la URL del repositorio y consérvela para la entrega.

### Ejemplo de estructura del repositorio

```text
actividad-pluma-agua/
│
├── README.md
├── pluma_agua.py
└── Actividad_Pluma_de_Agua_Algoritmo_GitHub.ipynb
```

## 11. Comandos Git sugeridos

> **Nota:** estos comandos deben ejecutarse en la terminal, Git Bash o PowerShell.

### 11.1 Inicializar el repositorio local

```bash
git init
git add .
git commit -m "Versión inicial de la actividad pluma de agua"
```

### 11.2 Conectar con GitHub

```bash
git branch -M main
git remote add origin URL_DEL_REPOSITORIO
git push -u origin main
```

### 11.3 Subir una segunda versión con mejor documentación

```bash
git add .
git commit -m "Se mejora documentación y comentarios del código"
git push
```

## 12. Clonar el repositorio en el disco local C:

Una vez que el repositorio esté en GitHub, copie su URL y ejecute:

```bash
cd /c
git clone URL_DEL_REPOSITORIO
```

Si está usando PowerShell en Windows, también puede hacerlo así:

```powershell
cd C:\
git clone URL_DEL_REPOSITORIO
```

### Ejemplo esperado

Si su repositorio se llama:

```text
https://github.com/usuario/actividad-pluma-agua.git
```

entonces el comando sería:

```bash
git clone https://github.com/usuario/actividad-pluma-agua.git
```

Con esto se creará una carpeta del proyecto dentro de `C:\`.

## 13. Evidencia de entrega sugerida

El estudiante debe entregar:

- el enlace del repositorio en GitHub;
- el notebook o archivo `.py` funcional;
- una versión documentada del código;
- evidencia de que el repositorio fue clonado en el equipo local.

### Evidencia mínima recomendada
- captura del repositorio en GitHub;
- captura de los commits;
- captura del comando `git clone` ejecutado correctamente.

## 14. Reflexión final

Este ejercicio no trata solamente de programar.  
También muestra cómo una palabra cotidiana del Caribe colombiano puede llevarnos a:

- la historia de la hidráulica;
- la representación computacional de una magnitud física;
- el diseño de algoritmos;
- el uso responsable de herramientas como GitHub y ChatGPT en el trabajo académico.

> La programación también puede nacer de la memoria cultural y del contexto local.

## 15. Espacio para notas del estudiante

Escriba aquí una breve reflexión personal:

- ¿Qué aprendió?
Aprendi a traducir un concepto historico en un modelo y codigo siguienddo un algortimo para dar con los resultados necesarios de la situacion problema
- ¿Qué parte le resultó más difícil?
la parte mas dificil fue escribir correctamente las variables y constantes sugeridas
- ¿Qué mejoró cuando documentó el código?me ayudo a entender e identificar las fallas que tuve en cada sección
- ¿Qué importancia tiene publicar y versionar su trabajo?
tener un registro de mi proceso y de  lo que aprendi es importante para poder mejorar mis habilidades en programacion e implementarlas en situaciones cotidianas